src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -757,6 +758,24 @@ def copy(self) -> "SnowflakeQueryCompiler":
         qc.snowpark_pandas_api_calls = self.snowpark_pandas_api_calls.copy()
         return qc
 
+    def to_list(self) -> list:
+        """
+        Return a native Python list of the values.
+
+        Only called if the frontend object was a Series.
+        """
+        return self.to_pandas().squeeze().to_list()
```

Comment:
this seems a wired implementation here, which seems tailored specifically for series. what is the modin implementation ? can we use the overwrite for this for now, and see how can we migrate over later?

Reply:
DataFrames don't have a `to_list` method. Upstream Modin does something similar:

```python
    def to_list(self):
        """
        Return a list of the values.

        These are each a scalar type, which is a Python scalar (for str, int, float) or a pandas scalar (for Timestamp/Timedelta/Interval/Period).

        Returns
        -------
        list
        """
        return SeriesDefault.register(pandas.Series.to_list)(self)
```

Reply:
I actualy see an override of to_list in the series_override methods, which seems kind of saying this method is not called. Since this is just a refactoring pr, i would suggest let's just use override for now, and handle those implementation change in a separate pr

Reply:
(discussed on slack) moved back to frontend overrides